### DISCUSSION TITLE
How to use scripts to conditionally alter a table?

### DISCUSSION THREAD

**Discussion by gtokman** [supabase/supabase#2423] [Questions]
I'm creating a web app that allows users to upvote posts. When a user upvotes the post, I want to check if they have upvoted the post before otherwise add the post id to their liked posts and increment how many likes the app has.

Users
| id      | liked_apps (int4[]) |
| ----------- | ----------- |
| 1   | [4]        |

Apps
| id      | like_count  |
| ----------- | ----------- |
| 2      |  5      |

Would it make more sense to query the liked apps on the client-side and check the id of the app the user is attempting to like and if it's not in there I create an update or is it best to do these checks and updates with a query?

```
// Check the liked apps array
// Does the current id of the app exist, if so return the like count
// Else add it to the array and increment the like count and return
```

**Comment by silentworks** [comment]
I think this will be a personal preference thing, you can do either of the solutions you have above. I would personally do this in a stored function as it would keep all the logic encapsulated in one place.

**Reply by gtokman** [reply]
Do you by chance know where I can learn how to create such a function, I don't have much experience with SQL.

**Reply by silentworks** [reply]
Yeah sure, there is a really good guide on this website https://www.postgresqltutorial.com/postgresql-create-function/.

You can then create your function inside of the Supabase UI from the `SQL` tab.
